org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/contentassist/SnippetCompletionProposal.java

```diff
@@ -311,6 +367,39 @@ private static CompletionItem getInterfaceSnippet(ICompilationUnit cu, Completio
 		return interfaceSnippetItem;
 	}
 
+	private static CompletionItem getRecordSnippet(SnippetCompletionContext scc, IProgressMonitor monitor) {
+		ICompilationUnit cu = scc.getCompilationUnit();
+		IJavaProject javaProject = cu.getJavaProject();
+		if (javaProject == null) {
+			return null;
+		}
+		String version = javaProject.getOption(JavaCore.COMPILER_COMPLIANCE, true);
+		if (JavaModelUtil.isVersionLessThan(version, JavaCore.VERSION_14)) {
```

Comment:
at least the version comparison is more consistent with the rest of the codebase